addon/src/services/page-title.ts

```diff
@@ -37,24 +68,32 @@ export default class PageTitleService extends Service {
   };
 
   constructor() {
+    // eslint-disable-next-line prefer-rest-params
     super(...arguments);
     this._validateExistingTitleElement();
 
-    let config = getOwner(this).resolveRegistration('config:environment');
-    if (config.pageTitle) {
-      ['separator', 'prepend', 'replace'].forEach((key) => {
-        if (!isEmpty(config.pageTitle[key])) {
-          this._defaultConfig[key] = config.pageTitle[key];
+    const config = getOwner(this).factoryFor('config:environment')?.class as
```

Comment:
This needs a note explaining why it's safe.

```suggestion
    // SAFETY: This is only valid because we cast it to a type which is optional
    // *and* whose key is optional. It may be *wrong*, but it is at least sound.
    const config = getOwner(this).factoryFor('config:environment')?.class as
```